Option to prevent rewriting tsconfig.json file

For people that don't care about keeping `files` in sync (for other IDE use) or just don't want us messing with their `tsconfig.json` file formatting, provide an option to disable tsconfig.json overwriting

* [x] Add some place to use this setting (the simplest location would be to use `tsconfig.json` as that file is available to `tsconfig.ts` (our code that rewrites `tsconfig.json`)
* [x] Add code to `tsconfig.ts` to use this option
* [x] Document option  [here](https://github.com/TypeStrong/atom-typescript/blob/master/docs/tsconfig.md) and optionally [here](https://github.com/TypeStrong/atom-typescript/blob/master/docs/faq.md)

This feature is fairly annoying for my project's setup and I'd be happy to implement this. I'm currently having some trouble getting the project to build; are their docs on this?

When I run grunt, I get a ton of errors like:

atom-typescript/lib/main/atom/typescriptGrammar.ts(183,53): error TS2503: Cannot find namespace 'ts'.
atom-typescript/lib/main/lang/core/project.ts(43,32): error TS2304: Cannot find name 'ts'.
atom-typescript/lib/main/lang/fixmyts/quickFixes/addClassMember.ts(112,13): error TS2322: Type '{ span: { start: any; length: number; }; newText: string; filePath: any; }' is not assignable to type 'Refactoring'.

--- a/docs/faq.md
+++ b/docs/faq.md
@@ -3,6 +3,10 @@
 ## I keep getting changes to tsconfig.json
 This is probably because of us keeping `files` updated with the `filesGlob` option. The reason why we do this is because the official `tsconfig.json` spec does not support `filesGlob`. Therefore we keep `files` in sync with the `filesGlob` so that your team mates can use whatever editor they prefer (sublime text, visual studio etc.).
 
+You can now disable this behavior by setting the `rewriteTsconfig` flag to `false` in your project's `tsconfig.json` under the `atom` key.
+
+[Further Details](https://github.com/TypeStrong/atom-typescript/blob/master/docs/tsconfig.md#atom)
+
 ## For really large projects atom-typescript gets slow
 If you have `tsconfig.json` in a folder that contains `node_modules`, atom-typescript might become slow (due to extensive file listing). Two possible fixes:
 * Move `tsconfig.json` into a sub folder e.g. `src`

--- a/docs/tsconfig.md
+++ b/docs/tsconfig.md
@@ -20,6 +20,8 @@ i.e. an empty JSON file at the *root* of your project :heart: This will be suffi
 * [`compileOnSave`](https://github.com/TypeStrong/atom-typescript/blob/master/docs/tsconfig.md#compileonsave) : Should AtomTS compile on save
 * [`buildOnSave`](https://github.com/TypeStrong/atom-typescript/blob/master/docs/tsconfig.md#buildonsave) : Should AtomTS build on save
 * [`scripts`](https://github.com/TypeStrong/atom-typescript/blob/master/docs/tsconfig.md#scripts) : Sometimes its useful to have post build scripts
+* [`atom`](https://github.com/TypeStrong/atom-typescript/blob/master/docs/tsconfig.md#rewriteTsconfig) : Configuration specific to Atom. Currently
+only contains `rewriteTsconfig` which prevents Atom from rewriting a project's `tsconfig.json`.
 
 
 ## Examples
@@ -112,6 +114,24 @@ Build means *compile all files*. Useful if for some reason you are using `--out`
 ```json
 {
   "buildOnSave": true
+}
+```
+
+### atom
+Configuration options specific to Atom.
+
+
+**rewriteTsconfig**
+
+Atom-Typescript constantly resolves the `filesGlob` listed in your `tsconfig.json` to ensure that the glob is in sync
+with your project. If your project doesn't require this (you are managing your `filesGlob` some other way), set this
+to `false` (this defaults to `true`).
+
+```json
+{
+  "atom": {
+    "rewriteTsconfig": true
+  }
 }
 ```
 

--- a/lib/main/tsconfig/tsconfig.ts
+++ b/lib/main/tsconfig/tsconfig.ts
@@ -129,6 +129,7 @@ interface TypeScriptProjectRawSpecification {
     buildOnSave?: boolean;
     externalTranspiler?: string | { name: string; options?: any };
     scripts?: { postbuild?: string };
+    atom?: { rewriteTsconfig?: boolean };
 }
 
 /**
@@ -146,6 +147,7 @@ export interface TypeScriptProjectSpecification {
     package?: UsefulFromPackageJson;
     externalTranspiler?: string | { name: string; options?: any };
     scripts: { postbuild?: string };
+    atom: { rewriteTsconfig: boolean };
 }
 
 ///////// FOR USE WITH THE API /////////////
@@ -337,7 +339,8 @@ export function getDefaultInMemoryProject(srcFile: string): TypeScriptProjectFil
         formatCodeOptions: formatting.defaultFormatCodeOptions(),
         compileOnSave: true,
         buildOnSave: false,
-        scripts: {}
+        scripts: {},
+        atom: { rewriteTsconfig: true },
     };
 
     return {
@@ -388,7 +391,7 @@ export function getProjectSync(pathOrSrcFile: string): TypeScriptProjectFileDeta
     if (projectSpec.filesGlob) { // for filesGlob we keep the files in sync
         var prettyJSONProjectSpec = prettyJSON(projectSpec, detectIndent(projectFileTextContent).indent);
 
-        if (prettyJSONProjectSpec !== projectFileTextContent) {
+        if (prettyJSONProjectSpec !== projectFileTextContent && projectSpec.atom.rewriteTsconfig) {
             fs.writeFileSync(projectFile, prettyJSONProjectSpec);
         }
     }
@@ -421,7 +424,8 @@ export function getProjectSync(pathOrSrcFile: string): TypeScriptProjectFileDeta
         typings: [],
         externalTranspiler: projectSpec.externalTranspiler == undefined ? undefined : projectSpec.externalTranspiler,
         scripts: projectSpec.scripts || {},
-        buildOnSave: !!projectSpec.buildOnSave
+        buildOnSave: !!projectSpec.buildOnSave,
+        atom: { rewriteTsconfig: true }
     };
 
     // Validate the raw compiler options before converting them to TS compiler options

